python_modules/libraries/dagster-airflow/dagster_airflow/hooks/dagster_hook.py

```diff
@@ -0,0 +1,252 @@
+      __typename
+    }
+    ... on InvalidSubsetError {
+      message
+      __typename
+    }
+    ... on RunConfigValidationInvalid {
+      errors {
+        message
+        __typename
+      }
+      __typename
+    }
+    ...PythonErrorFragment
+  }
+}
+
+fragment PythonErrorFragment on PythonError {
+  __typename
+  message
+  stack
+  causes {
+    message
+    stack
+    __typename
+  }
+}
+        """
+        variables = {
+            "executionParams":{
+                "runConfigData": json.dumps(run_config),
+                "selector":{
+                    "repositoryName": repository_name,
+                    "repositoryLocationName": repostitory_location_name,
+                    "jobName": job_name
+                },
+                "mode":"default",
+                "executionMetadata":{
+                    "tags":[
+                        {"key":"dagster/solid_selection","value":"*"}
+                    ]
+                }
+            }
+        }
+        headers = {'Dagster-Cloud-Api-Token': self.user_token}
+        response = requests.post(url=self.url, json={'query': query, 'variables': variables}, headers=headers)
+        response.raise_for_status()
+        response_json = response.json()
+        if response_json['data']['launchPipelineExecution']['__typename'] == 'LaunchRunSuccess':
+            run = response_json['data']['launchPipelineExecution']['run']
+            print(f"Run {run['id']} launched successfully")
+            return run['id']
+        else :
+            raise AirflowException(f'Error launching run: {response_json["data"]["launchPipelineExecution"]["message"]}')
+
+    def wait_for_run(
+        self,
+        run_id: str,
+    ) -> None:
+      query = """
+query RunQuery($runId: ID!) {
+	runOrError(runId: $runId) {
+		__typename
+		...PythonErrorFragment
+		...NotFoundFragment
+		... on Run {
+			id
+			status
+			__typename
+		}
+	}
+}
+fragment NotFoundFragment on RunNotFoundError {
+	__typename
+	message
+}
+fragment PythonErrorFragment on PythonError {
+	__typename
+	message
+	stack
+	causes {
+		message
+		stack
+		__typename
+	}
+}
+      """
+      variables = {
+          "runId": run_id
+      }
+      headers = {'Dagster-Cloud-Api-Token': self.user_token}
+      status = ''
+      while status not in [PipelineRunStatus.SUCCESS.value, PipelineRunStatus.FAILURE.value, PipelineRunStatus.CANCELED.value]:
+        response = requests.post(url=self.url, json={'query': query, 'variables': variables}, headers=headers)
+        response.raise_for_status()
+        response_json = response.json()
+
+        if response_json['data']['runOrError']['__typename'] == 'Run':
+            status = response_json['data']['runOrError']['status']
+        else :
+            raise AirflowException(f'Error fetching run status: {response_json["data"]["runOrError"]["message"]}')
+
+        if status == PipelineRunStatus.SUCCESS.value:
+          print(f"Run {run_id} completed successfully")
```

Comment:
out of curiosity, where would the prints go?

Reply:
they'll go into airflows log system